dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/cache/CacheEntry.java

```diff
@@ -302,7 +322,11 @@ private void readModel(Path input) throws IOException {
     }
 
     private void writeModel(Path output) throws IOException {
-        String jsonModel = OpenApiSerializer.serialize(model, Format.JSON);
+        String jsonModel;
+        // This method runs asynchronously, so synchronize to ensure we don't use the model at the same time as WebModuleOpenAPIProvider
+        synchronized (model) {
```

Comment:
This is an exclusive lock...but we really only need a Read lock?

Also when one adds a lock, one must always be careful if there is the possibility of trying
to acquire a second lock or waiting for something while holding it. In case another code
path might do the same in reverse - creating a hang. Can any other thread lock this
same model? If so the codepaths need to be checked and a common lock ordering firmly
established.

Reply:
Correct, we only need a read lock, but there's only two participants here and the other one needs a write lock.

I don't think any other locks are involved in either place that synchronises on the model but I will check.

Reply:
OK sounds good